From: Johannes Weiner <hannes@cmpxchg.org>
Subject: sched-loadavg-consolidate-load_int-load_frac-calc_load-fix-fix

The macro used to modify the avg parameter in place, but with the
function we need an explicit assignment to update the variable:

Link: http://lkml.kernel.org/r/20180913014923.GB2370@cmpxchg.org
Signed-off-by: Johannes Weiner <hannes@cmpxchg.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/block/blk-iolatency.c~sched-loadavg-consolidate-load_int-load_frac-calc_load-fix-fix
+++ a/block/blk-iolatency.c
@@ -512,7 +512,8 @@ static void iolatency_check_latencies(st
 	exp_idx = min_t(int, BLKIOLATENCY_NR_EXP_FACTORS - 1,
 			div64_u64(iolat->cur_win_nsec,
 				  BLKIOLATENCY_EXP_BUCKET_SIZE));
-	calc_load(iolat->lat_avg, iolatency_exp_factors[exp_idx], stat.mean);
+	iolat->lat_avg = calc_load(iolat->lat_avg,
+				   iolatency_exp_factors[exp_idx], stat.mean);
 
 	/* Everything is ok and we don't need to adjust the scale. */
 	if (stat.mean <= iolat->min_lat_nsec &&
_
